lib/discover/hyperbahn.js

```diff
@@ -0,0 +1,75 @@
+var TChannelAsThrift = require('tchannel/as/thrift');
+var TChannel = require('tchannel');
+var fs = require('fs');
+var path = require('path');
+
+module.exports.discover = function hyperbahnDiscover(urlObj, original, callback) {
+    var host;
+    if (urlObj.host) {
+        host = urlObj.host;
+    } else {
+        host = '127.0.0.1:21300';
+    }
+
+    var client = TChannel();
+    var hChannel = client.makeSubChannel({
+        serviceName: 'hyperbahn',
+        peers: [host]
+    });
+    var tchannelAsThrift = TChannelAsThrift({
+        channel: hChannel,
+        entryPoint: path.join(__dirname, 'hyperbahn.thrift')
+    });
+
+    tchannelAsThrift.waitForIdentified({
+        host: host
+    }, function onIdentified(err) {
+        if (err) {
+            callback(err);
+            return;
+        }
+
+        tchannelAsThrift.request({
+            host: host,
```

Comment:
Has been added, but github does not recognize the added line below as being part of this discussion.